feat: add `findIndex` method to `array/fixed-endian-factory` (#3239)
stdlib-js/stdlib: lib/node_modules/@stdlib/array/fixed-endian-factory/README.md

// returns undefined
```

<a name="method-find-index"></a>

> #### TypedArrayFE.prototype.findIndex( callbackFn\[, thisArg] )

Reviewer: ```suggestion
#### TypedArrayFE.prototype.findIndex( predicate\[, thisArg] )
```

Renaming to ensure consistent documentation across array types.

> Returns the index of the first element in a typed array that satisfies the provided testing function. 
> If no element satisfies function, return -1.

Reviewer: ```suggestion
Returns the index of the first element in a typed array for which a predicate function returns a truthy value.
```

To ensure consistent docs across array types.

```javascript

> function greaterthantwo( v ) {
>     return ( v > 2.0 );
> }
> 
> var Float64ArrayFE = fixedEndianFactory( 'float64' );
> 
> var arr = new Float64ArrayFE( 'little-endian', 3 );
> 
> arr.set( 1.5, 0 );
> arr.set( 2.5, 1 );
> arr.set( 3.5, 2 );
> 
> var idx = arr.findIndex( greaterthantwo );

Reviewer: ```suggestion
function predicate( v ) {
    return ( v > 2.0 );
}

var Float64ArrayFE = fixedEndianFactory( 'float64' );

var arr = new Float64ArrayFE( 'little-endian', 3 );

arr.set( 1.5, 0 );
arr.set( 2.5, 1 );
arr.set( 3.5, 2 );

var idx = arr.findIndex( predicate );
```

// returns 1
```

> The invoked function is provided three arguments:

Reviewer: ```suggestion
The `predicate` function is provided three arguments:
```

-   **value**: current array element.
-   **index**: current array element index.
-   **arr**: the array on which this method was called.

To set the function execution context, provide a `thisArg`.

```javascript

> function fcn( v, i ) {
>     this.count += 1;
>     return (v === 2.0);
> }
> 
> var Float64ArrayFE = fixedEndianFactory( 'float64' );
> 
> var arr = new Float64ArrayFE( 'little-endian', 3 );
> 
> var context = {
>     'count': 0
> };
> 
> arr.set( 1.0, 0 );
> arr.set( 2.0, 1 );
> arr.set( 3.0, 2 );
> 
> arr.findIndex( fcn, context );

Reviewer: ```suggestion
function predicate( v, i ) {
    this.count += 1;
    return ( v === 2.0 );
}

var Float64ArrayFE = fixedEndianFactory( 'float64' );

var arr = new Float64ArrayFE( 'little-endian', 3 );

var context = {
    'count': 0
};

arr.set( 1.0, 0 );
arr.set( 2.0, 1 );
arr.set( 3.0, 2 );

arr.findIndex( predicate, context );
```
